### PR TITLE
feat(pipelined): Router Advertisement issue fix for 5G UE IPv6

### PR DESCRIPTION
**Pull request by prabinakpattnaik**
Signed-off-by: prabina pattnaik <prabinak@wavelabs.ai>

## Summary
Router Advertisement not getting generated for 5G UE IPv6 [Issue ID #12871 ]

Below are functionality supported in this PR:
For 4g and 5g controller ID is different value used.
So, ipv6 solicitation flows are inserted into table 3 with Action value different based on cotroller ID
For 5g, output of table=3:
```
vagrant@magma-dev-focal:~/magma/lte/gateway$ sudo ovs-ofctl -O OpenFlow13 dump-flows gtp_br0 table=3
 cookie=0x0, duration=52.337s, table=3, n_packets=0, n_bytes=0, priority=10,icmp6,reg1=0x1,ipv6_src=fe80::/10,icmp_type=133 actions=controller(reason=action_set,max_len=0,id=5),CONTROLLER:65535
 cookie=0x0, duration=52.320s, table=3, n_packets=0, n_bytes=0, priority=10,icmp6,reg1=0x1,ipv6_src=fe80::/10,icmp_type=135 actions=controller(reason=action_set,max_len=0,id=5),CONTROLLER:65535
 cookie=0x0, duration=52.318s, table=3, n_packets=0, n_bytes=0, priority=10,icmp6,reg1=0x10,ipv6_dst=ff02::1:ff00:0/104,icmp_type=135 actions=controller(reason=action_set,max_len=0,id=5),CONTROLLER:65535
```
Need to add GNB IP address in "/etc/network/interfaces.d/gtp" at port gtp0 for sending the Router adverstisemnt message to target device:
```
        Port gtp0
            Interface gtp0
                type: gtpu
                options: {key=flow, remote_ip="192.168.60.20"}
```
## Test Plan
Using UERANSIM simulator, we have verified this functionality.

## Additional Information

Logs:
IP Address Allocated for IPv6:
```
Jun 17 20:39:21 magma-dev-focal mobilityd[183930]: INFO:root:Allocated IP fdee:5:6c:2db6:dcb0:b4af:481d:1f1 for sid IMSI901700000000001 for apn internet
Jun 17 20:39:21 magma-dev-focal policydb[184001]: INFO:root:Creating a session for subscriber ID: IMSI901700000000001
```
IPV6SolicitationController logs when received Router Solicitation from UE:
```
Jun 17 20:42:38 magma-dev-focal pipelined[184051]: DEBUG:magma.pipelined.openflow.flows:flowmod: version=None,msg_type=None,msg_len=None,xid=None,OFPFlowStatsRequest(cookie=0,cookie_mask=0,flags=0,match=OFPMatch(oxm_fields={}),out_group=4294967295,out_port=4294967295,table_id=14,type=1) (table 14)
Jun 17 20:42:38 magma-dev-focal pipelined[184051]: DEBUG:IPV6SolicitationController:Received PKT -> on port 32768 tun_id: 2147483647 tun_id_dst: 1
Jun 17 20:42:38 magma-dev-focal pipelined[184051]: DEBUG:IPV6SolicitationController:ethernet(dst='ff:ff:ff:ff:ff:ff',ethertype=34525,src='02:00:00:00:00:01')
Jun 17 20:42:38 magma-dev-focal pipelined[184051]: DEBUG:IPV6SolicitationController:ipv6(dst='ff02::2',ext_hdrs=[],flow_label=0,hop_limit=255,nxt=58,payload_length=8,src='fe80::dcb0:b4af:481d:1f1',traffic_class=0,version=6)
Jun 17 20:42:38 magma-dev-focal pipelined[184051]: DEBUG:IPV6SolicitationController:icmpv6(code=0,csum=41416,data=nd_router_solicit(option=None,res=0),type_=133)
Jun 17 20:42:38 magma-dev-focal pipelined[184051]: DEBUG:IPV6SolicitationController:Received router solicitation MSG
Jun 17 20:42:38 magma-dev-focal pipelined[184051]: DEBUG:IPV6SolicitationController:RA pkt response ->
Jun 17 20:42:38 magma-dev-focal pipelined[184051]: DEBUG:IPV6SolicitationController:ethernet(dst='33:33:00:00:00:01',ethertype=34525,src='08:00:27:eb:e7:39')
Jun 17 20:42:38 magma-dev-focal pipelined[184051]: DEBUG:IPV6SolicitationController:ipv6(dst='fe80::dcb0:b4af:481d:1f1',ext_hdrs=[],flow_label=0,hop_limit=255,nxt=58,payload_length=0,src='fe80::a00:27ff:fe90:db2c',traffic_class=0,version=6)
Jun 17 20:42:38 magma-dev-focal pipelined[184051]: DEBUG:IPV6SolicitationController:icmpv6(code=0,csum=0,data=nd_router_advert(ch_l=255,options=[nd_option_sla(data=None,hw_src='08:00:27:eb:e7:39',length=0), nd_option_pi(length=0,pl=64,pre_l=4294967295,prefix='fdee:5:6c:2db6::',res1=6,res2=0,val_l=4294967295)],rea_t=0,res=0,ret_t=0,rou_l=65535),type_=134)
```

### CODE REVIEW

File: lte/gateway/python/magma/pipelined/app/ipv6_solicitation.py [hunk 1]
```diff
@@ -90,11 +91,15 @@ def _get_config(self, config_dict):
             ipv6_src = get_link_local_ipv6_from_if(s1_interface)
             if ipv6_src is None:
                 ipv6_src = self.DEFAULT_LINK_LOCAL_ADDR
-
+        ng_service = config_dict.get('enable5g_features', None)
+        logging.info("enable5g")
```
Comment (nstng) [inline lte/gateway/python/magma/pipelined/app/ipv6_solicitation.py]: Can we merge these two logging lines and make them more descriptive? Is the log-level appropriate?

File: lte/gateway/python/magma/pipelined/app/ipv6_solicitation.py [hunk 2]
```diff
@@ -129,6 +134,13 @@ def _install_default_ipv6_flows(self, datapath):
             direction=Direction.OUT,
         )
 
+        if self.config.ng_service_enabled == True:
+            action = [
```
Comment (nstng) [inline lte/gateway/python/magma/pipelined/app/ipv6_solicitation.py:134]: Is there a more descriptive name for this variable? Nit: should be plural, it is a list. Or the controller is created here and used as actions=[controller].

File: lte/gateway/python/magma/pipelined/app/ipv6_solicitation.py [hunk 3]
```diff
@@ -129,6 +134,13 @@ def _install_default_ipv6_flows(self, datapath):
             direction=Direction.OUT,
         )
 
+        if self.config.ng_service_enabled == True:
```
Comment (nstng) [inline lte/gateway/python/magma/pipelined/app/ipv6_solicitation.py]: Is the order of the flows important here? Looks strange to have this check multiple times. Can this be reordered and become guarded by one if-statement?

File: lte/gateway/python/magma/pipelined/app/ipv6_solicitation.py [hunk 4]
```diff
@@ -90,11 +91,15 @@ def _get_config(self, config_dict):
             ipv6_src = get_link_local_ipv6_from_if(s1_interface)
             if ipv6_src is None:
                 ipv6_src = self.DEFAULT_LINK_LOCAL_ADDR
-
+        ng_service = config_dict.get('enable5g_features', None)
```
Comment (nstng) [inline lte/gateway/python/magma/pipelined/app/ipv6_solicitation.py]: Why not default to `False`? Can also be named `ng_service_enabled`

File: lte/gateway/python/magma/pipelined/app/ipv6_solicitation.py (hunk 3 above)
Comment (nstng) [inline lte/gateway/python/magma/pipelined/app/ipv6_solicitation.py]: If this is sure to be boolean (see above), then the `== True` should be redundant.

File: lte/gateway/python/magma/pipelined/app/ipv6_solicitation.py [hunk 5]
```diff
@@ -90,11 +91,15 @@ def _get_config(self, config_dict):
...
+        logging.info("enable5g")
+        logging.info(ng_service)
         return self.IPv6RouterConfig(
             ipv6_src=ipv6_src,
             ll_addr=config_dict['virtual_mac'],
             prefix_len=self.DEFAULT_PREFIX_LEN,
+            ng_service_enabled=ng_service,
```
Comment (nstng) [inline lte/gateway/python/magma/pipelined/app/ipv6_solicitation.py]: `classifier_controller_id` is only used if `ng_service_enabled` is true. Should these configs be merged?

Comment (nstng) [inline lte/gateway/python/magma/pipelined/app/ipv6_solicitation.py]: `ng_service_enabled` True is not tested in `test_ipv6_solicitation`. Does a test make sense here? What is expected if it is True, but `classifier_controller_id` is not set?

File: lte/gateway/python/magma/pipelined/tests/test_ipv6_solicitation.py [hunk 6]
```diff
@@ -106,6 +106,7 @@ def setUpClass(cls, *_):
                 'clean_restart': True,
                 'virtual_mac': 'd6:34:bc:81:5d:40',
                 'enable_nat': True,
+                'classifier_controller_id': 5,
```
Comment (nstng) [inline lte/gateway/python/magma/pipelined/tests/test_ipv6_solicitation.py:109]: Why is `classifier_controller_id` required if `ng_service_enabled` is not set?

**Review by nstng — COMMENTED** [review]
Will continue when first set of comments is addressed.

File: lte/gateway/python/magma/pipelined/app/ipv6_solicitation.py (hunk 1 above)
Comment (prabinakpattnaik) [reply lte/gateway/python/magma/pipelined/app/ipv6_solicitation.py]: Yes, forget to remove the logging lines.
Fixed it.

File: lte/gateway/python/magma/pipelined/app/ipv6_solicitation.py (hunk 2 above)
Comment (prabinakpattnaik) [reply lte/gateway/python/magma/pipelined/app/ipv6_solicitation.py:134]: Changes done.

File: lte/gateway/python/magma/pipelined/app/ipv6_solicitation.py (hunk 3 above)
Comment (prabinakpattnaik) [reply lte/gateway/python/magma/pipelined/app/ipv6_solicitation.py]: Changes done.

File: lte/gateway/python/magma/pipelined/app/ipv6_solicitation.py (hunk 4 above)
Comment (prabinakpattnaik) [reply lte/gateway/python/magma/pipelined/app/ipv6_solicitation.py]: fixed it.

File: lte/gateway/python/magma/pipelined/app/ipv6_solicitation.py (hunk 5 above)
Comment (prabinakpattnaik) [reply lte/gateway/python/magma/pipelined/app/ipv6_solicitation.py]: Fixed it.

File: lte/gateway/python/magma/pipelined/app/ipv6_solicitation.py (hunk 3 above)
Comment (prabinakpattnaik) [reply lte/gateway/python/magma/pipelined/app/ipv6_solicitation.py]: Yes, changes done

File: lte/gateway/python/magma/pipelined/app/ipv6_solicitation.py (hunk 5 above)
Comment (prabinakpattnaik) [reply lte/gateway/python/magma/pipelined/app/ipv6_solicitation.py]: The ```classifier_controller_id``` value set to 5 in pipelined.yml.
This value used for 5G RYU controller set value for packetIN events.

File: lte/gateway/python/magma/pipelined/tests/test_ipv6_solicitation.py (hunk 6 above)
Comment (prabinakpattnaik) [reply lte/gateway/python/magma/pipelined/tests/test_ipv6_solicitation.py:109]: We should be defined all variable which are used in a code otherwise getting Exception error like as below:
```
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/usr/lib/python3.8/threading.py", line 932, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.8/threading.py", line 870, in run
    self._target(*self._args, **self._kwargs)
  File "/home/vagrant/magma/lte/gateway/python/magma/pipelined/tests/app/start_pipelined.py", line 238, in start_ryu_apps
    services.extend(manager.instantiate_apps(**contexts))
  File "/home/vagrant/build/python/lib/python3.8/site-packages/ryu/base/app_manager.py", line 509, in instantiate_apps
    self._instantiate(app_name, cls, *args, **kwargs)
  File "/home/vagrant/build/python/lib/python3.8/site-packages/ryu/base/app_manager.py", line 495, in _instantiate
    app = cls(*args, **kwargs)
  File "/home/vagrant/magma/lte/gateway/python/magma/pipelined/app/ipv6_solicitation.py", line 78, in __init__
    self.config = self._get_config(kwargs['config'])
  File "/home/vagrant/magma/lte/gateway/python/magma/pipelined/app/ipv6_solicitation.py", line 98, in _get_config
    classifier_controller_id=config_dict['classifier_controller_id'],
KeyError: 'classifier_controller_id'
E
```

File: lte/gateway/python/magma/pipelined/app/classifier.py [hunk 7]
```diff
@@ -82,7 +84,13 @@ def __init__(self, *args, **kwargs):
...
         self.next_table = self._service_manager.get_table_num(INGRESS)
         self.loop = kwargs['loop']
-        self._uplink_port = OFPP_LOCAL
+        self.uplink_port_name = kwargs['config']['ovs_uplink_port_name']
+        if self.config.enable_nat is False and self.uplink_port_name is not None:
+            self._uplink_port = BridgeTools.get_ofport(
+                self.uplink_port_name,
+            )
+        else:
+            self._uplink_port = OFPP_LOCAL
```
Comment (pshelar) [inline lte/gateway/python/magma/pipelined/app/classifier.py]: This PR is trying to implement two different functionality.
1. Non-NAT
2. Add support for IPv6.

Please separate it in two separate PRs with its own unit tests.

**Review by pshelar — CHANGES_REQUESTED** [review]

File: lte/gateway/python/magma/pipelined/app/classifier.py (hunk 7 above)
Comment (prabinakpattnaik) [reply lte/gateway/python/magma/pipelined/app/classifier.py]: @pshelar , we have fixed it as per your suggestion.
1) Created PR Support for Non-NAT mode in 5G-SA #13288  and merged into master.
2) Rebased with master and added UT for ipv6 solicitation
python/magma/pipelined/tests/test_ipv6_solicitation_5g.py

We have verified with UERANSIM for below functionality and along with end-to-end traffic flows:
Router Solicitation
Router Advertisement
IPv6 traffic with TCP.

Attached logs and pcap file for same.

[tc2.txt](https://github.com/magma/magma/files/9132581/tc2.txt)
[IPv6_Traffic_flows.zip](https://github.com/magma/magma/files/9132655/IPv6_Traffic_flows.zip)

**Review by pshelar — APPROVED** [review]

**Comment by maxhbr** [comment]
please rebase and fix `Bazel Build & Test`

File: lte/gateway/python/magma/pipelined/tests/test_ipv6_solicitation_5g.py [hunk 8]
```diff
@@ -0,0 +1,185 @@
+"""
+Copyright 2020 The Magma Authors.
```
Comment (nstng) [inline lte/gateway/python/magma/pipelined/tests/test_ipv6_solicitation_5g.py]: NIT: year

File: lte/gateway/python/magma/pipelined/app/classifier.py [hunk 9]
```diff
@@ -363,8 +363,8 @@ def _install_downlink_arp_flows(
 
     def add_tunnel_flows(
         self, precedence: int, i_teid: int,
-        o_teid: int, enodeb_ip_addr: Union[str, IPAddress],
-        ue_ip_adr: Optional[IPAddress] = None, sid: Optional[int] = None,
+        o_teid: int, enodeb_ip_addr: str,
```
Comment (nstng) [inline lte/gateway/python/magma/pipelined/app/classifier.py]: why were the type annotations removed? (also line below)

File: lte/gateway/python/magma/pipelined/tests/test_ipv6_solicitation_5g.py [hunk 10]
```diff
@@ -0,0 +1,185 @@
...
+    @classmethod
+    def tearDownClass(cls):
+        stop_ryu_app_thread(cls.thread)
+        BridgeTools.destroy_bridge(cls.BRIDGE)
+
+    def test_ipv6_5g_flows(self):
+        """
+        Verify that a UPLINK->UE Router Solic & Router Adv
+        """
+        ll_addr = get_if_hwaddr('testing_br')
```
Comment (nstng) [inline lte/gateway/python/magma/pipelined/tests/test_ipv6_solicitation_5g.py:138]: starting here, down to line 164 does not seem to be needed for the test. Also respective imports and class variables.

File: lte/gateway/python/magma/pipelined/tests/test_ipv6_solicitation_5g.py [hunk 11]
```diff
@@ -0,0 +1,185 @@
...
+                'quota_check_ip': '1.2.3.4',
+                'ipv6_router_addr': 'd88d:aba4:472f:fc95:7e7d:8457:5301:ebce',
+                'clean_restart': True,
+                'virtual_mac': 'd6:34:bc:81:5d:40',
+                'enable_nat': True,
+                'classifier_controller_id': 5,
+                'enable5g_features': True,
+            },
+            mconfig=PipelineD(
+                ue_ip_block=cls.UE_BLOCK,
```
Comment (nstng) [inline lte/gateway/python/magma/pipelined/tests/test_ipv6_solicitation_5g.py]: does not seem to be needed for the test

File: lte/gateway/python/magma/pipelined/rpc_servicer.py [hunk 12]
```diff
@@ -1045,6 +1045,7 @@ def _ng_qer_update(
                 )
             if qos_enforce_rule.ipv6_addr:
                 ipv6 = convert_ipv6_bytes_to_ip_proto(qos_enforce_rule.ipv6_addr)
+                self._update_ipv6_prefix_store(qos_enforce_rule.ipv6_addr)
```
Comment (nstng) [inline lte/gateway/python/magma/pipelined/rpc_servicer.py:1048]: just curious, why is the store update not needed in the `PdrState.Value('IDLE')` case?

File: lte/gateway/python/magma/pipelined/rpc_servicer.py [hunk 13]
```diff
@@ -987,7 +987,7 @@ def _ng_tunnel_update(self, pdr_entry: PDRRuleEntry, subscriber_id: str) -> bool
             pdr_entry.pdr_state,
             pdr_entry.precedence,
             pdr_entry.local_f_teid,
-            pdr_entry.far_action.o_teid,
+            pdr_entry.o_teid,
```
Comment (nstng) [inline lte/gateway/python/magma/pipelined/rpc_servicer.py:990]: are the changes in this file (and I assume related `session_state_manager_util.py)` covered by a test?

**Review by nstng — COMMENTED** [review]
Bazel changes in general lgtm, left some general comments for the other changes.
(feel free, to request a re-review via github, then I get a notification that I can have look)

**Comment by maxhbr** [comment]
please rebase, `please rebase` is still red

**Comment by prabinakpattnaik** [comment]
@maxhbr , Three not required checks are failing. I think it is not a red

**Comment by maxhbr** [comment]
it is red, so your PR is not fully rebased. Can you fix that please?

File: lte/gateway/python/magma/pipelined/tests/test_ipv6_solicitation_5g.py (hunk 8 above)
Comment (prabinakpattnaik) [reply lte/gateway/python/magma/pipelined/tests/test_ipv6_solicitation_5g.py]: Yes, fixed it.

File: lte/gateway/python/magma/pipelined/app/classifier.py (hunk 9 above)
Comment (prabinakpattnaik) [reply lte/gateway/python/magma/pipelined/app/classifier.py]: Yes, might be issue in merging. Fixed it.

File: lte/gateway/python/magma/pipelined/rpc_servicer.py (hunk 12 above)
Comment (prabinakpattnaik) [reply lte/gateway/python/magma/pipelined/rpc_servicer.py:1048]: Yes, it's required but for idle case fully not supported or verified. Once will start the work for Idle mode for IPv6 then will add it.

File: lte/gateway/python/magma/pipelined/tests/test_ipv6_solicitation_5g.py (hunk 11 above)
Comment (prabinakpattnaik) [reply lte/gateway/python/magma/pipelined/tests/test_ipv6_solicitation_5g.py]: Yes, but need to defined one mconfig parameter otherwise getting below error:
TypeError: __init__() missing 1 required positional argument: 'mconfig'

**Comment by prabinakpattnaik** [comment]
@pshelar , @maxhbr , now all required checks are passed and fixed the rebase check as well. Could you please push the changes into master.

File: lte/gateway/python/magma/pipelined/tests/test_ipv6_solicitation_5g.py (hunk 10 above)
Comment (maxhbr) [reply lte/gateway/python/magma/pipelined/tests/test_ipv6_solicitation_5g.py:138]: @prabinakpattnaik : can you answer here?

File: lte/gateway/python/magma/pipelined/tests/test_ipv6_solicitation_5g.py (hunk 11 above)
Comment (maxhbr) [reply lte/gateway/python/magma/pipelined/tests/test_ipv6_solicitation_5g.py]: ~~maybe @nstng was saying that `mconfig` is unused. Does `PipelineD(..)` have side effects?~~, ah, this is a parameter to `TestSetup`, the long function call confused me.

File: lte/gateway/python/magma/pipelined/tests/test_ipv6_solicitation_5g.py (hunk 10 above)
Comment (maxhbr) [reply lte/gateway/python/magma/pipelined/tests/test_ipv6_solicitation_5g.py:138]: the variables `pkt_rs` and `pkt_ns` are used in lines 179f. It also registers a package injector on a interface defined in setup, that is used later. For me that code looks needed.

File: lte/gateway/python/magma/pipelined/tests/test_ipv6_solicitation_5g.py (hunk 11 above)
Comment (nstng) [reply lte/gateway/python/magma/pipelined/tests/test_ipv6_solicitation_5g.py]: The `ue_ip_block` argument can be removed - not the complete `mconfig` argument.

I will close conversation anyway and move further discussion to the general test simplification discussion.

File: lte/gateway/python/magma/pipelined/tests/test_ipv6_solicitation_5g.py (hunk 10 above)
Comment (nstng) [reply lte/gateway/python/magma/pipelined/tests/test_ipv6_solicitation_5g.py:138]: Here is a simplified version that is green for me: https://github.com/magma/magma/commit/5bc798c17430a6abcbb7ae0c7ce2b053aa0fc7f1

my understanding: the tests uses `include_stats=False`, i.e., it does not matter if any packages are sent because they are not verified in the rule statistics.

Comment (maxhbr) [reply lte/gateway/python/magma/pipelined/tests/test_ipv6_solicitation_5g.py:138]: ok, so this test does not test that much.

Comment (maxhbr) [reply lte/gateway/python/magma/pipelined/tests/test_ipv6_solicitation_5g.py:138]: The same is true for the test you copied the code from, due to the following line https://github.com/magma/magma/blob/a6d697122fba15843129109f3410f89a0e9477f8/lte/gateway/python/magma/pipelined/tests/test_ipv6_solicitation.py#L173

I think this needs a follow-up ticket where we resolve the tests which have unused code / test less then expected. I created https://github.com/magma/magma/issues/13616 for that

Comment (prabinakpattnaik) [reply lte/gateway/python/magma/pipelined/tests/test_ipv6_solicitation_5g.py:138]: I have removed the ```include_stats=False,``` for IPv6 traffic verification along with flows. It's not a dead code.

File: lte/gateway/python/magma/pipelined/tests/test_ipv6_solicitation_5g.py (hunk 11 above)
Comment (prabinakpattnaik) [reply lte/gateway/python/magma/pipelined/tests/test_ipv6_solicitation_5g.py]: Yes, we can remove the ue_ip_block argument but it's not a functionality issue.
Ok, I removed it for now.

File: lte/gateway/python/magma/pipelined/rpc_servicer.py (hunk 13 above)
Comment (prabinakpattnaik) [reply lte/gateway/python/magma/pipelined/rpc_servicer.py:990]: Yes,  it's functionality verification done.

File: lte/gateway/python/magma/pipelined/tests/test_ipv6_solicitation_5g.py (hunk 10 above)
Comment (nstng) [reply lte/gateway/python/magma/pipelined/tests/test_ipv6_solicitation_5g.py:138]: In the non 5g case this was set to False because of test instability (see respective PR). I did run the test locally 10 times -> all green. But this is of course no guarantee for CI.

File: lte/gateway/python/magma/pipelined/rpc_servicer.py (hunk 13 above)
Comment (nstng) [reply lte/gateway/python/magma/pipelined/rpc_servicer.py:990]: I do not understand what you mean. I do not see a new test for these changes.

**Review by nstng — APPROVED** [review]
Bazel changes lgtm

**Review by GANESH-WAVELABS — APPROVED** [review]
Sessiond Changes LGTM

File: lte/gateway/python/magma/pipelined/rpc_servicer.py (hunk 13 above)
Comment (prabinakpattnaik) [reply lte/gateway/python/magma/pipelined/rpc_servicer.py:990]: Only Variable name and place was changed. Remining are same as functionality. We have already verified the functionality with simulator.

File: lte/gateway/python/magma/pipelined/tests/test_ipv6_solicitation_5g.py (hunk 10 above)
Comment (prabinakpattnaik) [reply lte/gateway/python/magma/pipelined/tests/test_ipv6_solicitation_5g.py:138]: For traffic verification, it's required for True. Many TCs are present with out ```include_stats=False,``` for traffic verification.

**Review by Koki1323 — APPROVED** [review]

File: lte/gateway/python/magma/pipelined/tests/test_ipv6_solicitation_5g.py (hunk 10 above)
Comment (nstng) [reply lte/gateway/python/magma/pipelined/tests/test_ipv6_solicitation_5g.py:138]: @prabinakpattnaik example for the instability issue in CI for the 5g case: https://github.com/magma/magma/runs/7907532515?check_suite_focus=true
